x-pack/plugins/synthetics/public/apps/synthetics/contexts/synthetics_refresh_context.tsx

```diff
@@ -24,14 +24,21 @@ export const SyntheticsRefreshContext = createContext(defaultContext);
 export const SyntheticsRefreshContextProvider: React.FC = ({ children }) => {
   const [lastRefresh, setLastRefresh] = useState<number>(Date.now());
 
-  const refreshApp = () => {
+  const refreshApp = useCallback(() => {
     const refreshTime = Date.now();
     setLastRefresh(refreshTime);
-  };
+  }, [setLastRefresh]);
 
   const value = useMemo(() => {
     return { lastRefresh, refreshApp };
-  }, [lastRefresh]);
+  }, [lastRefresh, refreshApp]);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      refreshApp();
+    }, 1000 * 30);
+    return () => clearInterval(interval);
+  }, [refreshApp]);
```

Comment:
i think ideally this should be controlled from UI, i wonder if all the users will want this functionality in there browsers? i mean if let's say some users wants to disable it, do we want to provide that? 

we can perhaps user this component https://elastic.github.io/eui/#/templates/super-date-picker#auto-refresh 

with prop, `isAutoRefreshOnly: true`, seems ideal for this.

Reply:
👍 I agree this isn't ideal, but I'm reticent to add functionality in this case as it's supposed to be a bug fix.

We already have this functionality on the Overview page, so this extends the same functionality to the entire UI; it doesn't change the existing precedent that it's a hardcoded auto-refresh in our app though.

Reply:
We decided offline that we should create a follow-up enhancement ticket that enables the user to control this aspect of the UX.